tests/nested/manual/core20-factory-reset-install-device-hook/task.yaml

```diff
@@ -0,0 +1,181 @@
+    tests.nested build-image core
+    unset NESTED_FAKESTORE_SNAP_DECL_PC_GADGET
+
+    tests.nested create-vm core
+
+restore: |
+    if [ "$TRUST_TEST_KEYS" = "false" ]; then
+        echo "This test needs test keys to be trusted"
+        exit
+    fi
+
+    # stop fake device svc
+    systemctl stop fakedevicesvc
+
+    #shellcheck source=tests/lib/store.sh
+    . "$TESTSLIB"/store.sh
+    teardown_fake_store "$NESTED_FAKESTORE_BLOB_DIR"
+
+execute: |
+    if [ "$TRUST_TEST_KEYS" = "false" ]; then
+        echo "This test needs test keys to be trusted"
+        exit
+    fi
+
+    # wait for the initialize device task to be done
+    retry -n 200 --wait 1 sh -c "tests.nested exec snap changes | MATCH 'Done.*Initialize device'"
+
+    VERSION="$(tests.nested show version)"
+    BOOT_ID="$(tests.nested boot-id)"
+
+    echo "Check we have the right model from snap model"
+    tests.nested exec "sudo snap model --verbose" | MATCH "model:\s+testkeys-snapd-secured-core-$VERSION-amd64"
+    tests.nested exec "sudo snap model --verbose" | MATCH "grade:\s+secured"
+    tests.nested exec "sudo snap model --verbose --serial" | MATCH "serial:\s+7777"
+
+    echo "Check that the system-files interface is connected for the pc snap"
+    tests.nested exec "snap connections pc" | MATCH 'system-files\s+pc:modprobe-conf\s+:system-files'
+    tests.nested exec "snap connections pc" | MATCH 'system-files\s+pc:modules-load-conf\s+:system-files'
+    tests.nested exec "snap connections pc" | MATCH 'system-files\s+pc:udev-rules-conf\s+:system-files'
+
+    echo "Check that the directories have the right permissions"
+    tests.nested exec "stat /etc/modprobe.d/ -c %a" | MATCH 755
+    tests.nested exec "stat /etc/modules-load.d/ -c %a" | MATCH 755
+    tests.nested exec "stat /etc/udev/rules.d/ -c %a" | MATCH 755
+
+    echo "Check that the files from install-device were installed (install)"
+    tests.nested exec "cat /etc/modprobe.d/my-modprobe.conf" | MATCH '# configure modprobe here'
+    tests.nested exec "cat /etc/modules-load.d/my-modules-load.conf" | MATCH '# load modules here'
+    not tests.nested exec "stat /etc/udev/rules.d/09-my-custom-udev.rules"
+
+    # add || true in case the SSH connection is broken while executing this
+    # since this command causes an immediate reboot
+    tests.nested exec "sudo snap reboot --factory-reset" || true
+    tests.nested wait-for reboot "${BOOT_ID}"
+
+    # wait for the system to get setup and finish seeding
+    tests.nested wait-for snap-command
+    retry -n 10 --wait 2 tests.nested exec "sudo snap wait system seed.loaded"
+
+    # wait up to two minutes for serial registration
+    retry -n 60 --wait 2 tests.nested exec snap model --serial
+
+    # check that we are back in run mode
+    tests.nested exec cat /proc/cmdline | MATCH 'snapd_recovery_mode=run'
+
+    echo "Check that the files from install-device are installed (factory-reset)"
+    not tests.nested exec "cat /etc/modprobe.d/my-modprobe.conf"
```

Comment:
(mother-of-niticks) this uses `cat` and abvove `stat` to check if a file is missing :)

Reply:
Haha great catch! I can see that I was a bit to quick there, changing them to `stat`